### PR TITLE
LPS-182339 Mapping Link to Page to Fragment

### PR DESCRIPTION
**Pull request by YurenaC**
Motivation
=======
As a Customer, I want to map the Link to Page field in a fragment when I create a WebContent and try to map a fragment with a structure that contains a Link to Page field

Proposed Solution
========
Modify current code to allow that mapping

Steps to Verify
========

1. Go to Content & Data ->Structure and add a Link to Page field.
2. Create Web Content from the structure.
3. Go to Site Builder --> Pages and add a content (blank) page.
4. While editing the Page add a text fragment to it.
5. Click on the element of the fragment and on the Mapping/URL panel select the web content and as the Item and at the Field try to find the Content Structure.
6. The "link to Page" structure field should be available when mapping the fragment.

COMMITS:
- LPS-182339 Add link to layout to _SELECTABLE_DDM_STRUCTURE_FIELDS
- LPS-182339 Add LINK_TO_LAYOUT to getInfoType
- LPS-182339 Sanitize LINK_TO_LAYOUT value

### CODE REVIEW

**Comment by liferay-echo** [comment]
Thank you for submitting this PR. 👏

Make sure it follows the [Echo Pull Request Template](https://gist.github.com/liferay-echo/5a4455285ff1800f9da146816d85e67d#file-pull-request-template-md) if you haven't already done so. 

Learn more about it [here](https://liferay.atlassian.net/wiki/spaces/ENGECHO/pages/2027455014/Echo+Pull+Request+Etiquette).

**Comment by YurenaC** [comment]
ci:test:sf

**Comment by YurenaC** [comment]
ci:test:relevant

File: modules/apps/dynamic-data-mapping/dynamic-data-mapping-service/src/main/java/com/liferay/dynamic/data/mapping/internal/info/item/provider/DDMFormValuesInfoFieldValuesProviderImpl.java [hunk 1]
```diff
@@ -352,6 +352,30 @@ private Object _sanitizeDDMFormFieldValue(
...
+				JSONObject jsonObject = _jsonFactory.createJSONObject(
+					valueString);
+
+				long layoutId = jsonObject.getLong("layoutId");
+
+				if (layoutId == 0) {
+					return new ContentFieldValue();
+				}
+
+				Layout layoutByUuidAndGroupId = _layoutLocalService.getLayout(
```
Comment (ealonso) [inline modules/apps/dynamic-data-mapping/dynamic-data-mapping-service/src/main/java/com/liferay/dynamic/data/mapping/internal/info/item/provider/DDMFormValuesInfoFieldValuesProviderImpl.java:372]: @YurenaC the name should be layout, and if we have a fetch method we should use it and prevent null pointer exceptions, also, you should use the locale to get the friendlyURL

**Comment by YurenaC** [comment]
ci:stop

**Pull request closed without merging**